dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/deduplication/DeduplicationController.java

```diff
@@ -149,30 +152,33 @@ public void mergePotentialDuplicate(
       throws NotFoundException,
           PotentialDuplicateConflictException,
           PotentialDuplicateForbiddenException,
-          ForbiddenException,
-          BadRequestException {
+          ForbiddenException {
     PotentialDuplicate potentialDuplicate = deduplicationService.getPotentialDuplicate(uid);
 
     if (potentialDuplicate.getOriginal() == null || potentialDuplicate.getDuplicate() == null) {
       throw new PotentialDuplicateConflictException(
           "PotentialDuplicate is missing references and cannot be merged.");
     }
 
-    TrackedEntity original =
-        trackedEntityService.getTrackedEntity(potentialDuplicate.getOriginal());
-    TrackedEntity duplicate =
-        trackedEntityService.getTrackedEntity(potentialDuplicate.getDuplicate());
+    trackedEntityService.getNewTrackedEntity(potentialDuplicate.getOriginal());
+    trackedEntityService.getNewTrackedEntity(potentialDuplicate.getDuplicate());
 
     if (mergeObject == null) {
       mergeObject = new MergeObject();
     }
 
+    // TODO(tracker) jdbc-hibernate: check the impact on performance
+    TrackedEntity hibernateOriginal =
+        manager.get(TrackedEntity.class, potentialDuplicate.getOriginal());
+    TrackedEntity hibernateDuplicate =
+        manager.get(TrackedEntity.class, potentialDuplicate.getDuplicate());
+
     DeduplicationMergeParams params =
         DeduplicationMergeParams.builder()
             .potentialDuplicate(potentialDuplicate)
             .mergeObject(mergeObject)
-            .original(original)
-            .duplicate(duplicate)
+            .original(hibernateOriginal)
```

Comment:
is this a temporary fix only?

Reply:
Yes, in this case we should completely refactor deduplication service and use the importer instead of custom stores. And then, we wouldn't need to get the tracked entities using the manager.